File: npm/packages/@tailjs/types/src/TrackingSettings.ts
import type { UserInteractionEvent } from ".";
export interface TrackingSettings {
  /**
   * Always include in {@link UserInteractionEvent.components}, also if it is a parent component.
   * By default only the closest component will be included.
   *
   * This does not apply to impression tracking.
   *
   * Not inherited by child components.
   *
   * HTML attribute: `track-promote`.
   * CSS: `--track-promote: 1/yes/true`.
   *
   * @default false
   */
  promote?: boolean;

  /**
   * The component will only be tracked with the closest non-secondary component as if the latter had the {@link promote} flag.
   *
   * This does not apply to impression tracking.
   *
   * Not inherited by child components.
   *
   * HTML attribute: `track-secondary`. \
   * CSS: `--track-secondary: 1/yes/true`.
   *
   * @default false
   */
  secondary?: boolean;

  /**
   * Track the visible region occupied by the component or content.
   *
   * Inherited by child components (also if specified on non-component DOM element).
   *
   * HTML attribute: `track-region`. \
   * CSS: `--track-region: 1/yes/true`.
   *
   * @default false
   */
  region?: boolean;

  /**
   * Track clicks. Note that clicks are always tracked if they cause navigation.
   *
   * Inherited by child components (also if specified on non-component DOM element).
   *
   * HTML attribute: `track-clicks`.
   * CSS: `--track-clicks: 1/yes/true`.
   *
   * @default true unless in a `<nav>` tag
   */
  clicks?: boolean;

  /**
   * Track impressions, that is, when the component becomes visible in the user's browser for the first time.
   * This goes well with {@link region}.
   *
   * Not inherited by child components.
   *
   * HTML attribute: `track-impressions`.
   * CSS: `--track-impressions: 1/yes/true`.
   *
   * @default false
   */
  impressions?: boolean;
}

File: hosts/dotnet/core/TailJs/Model/TrackingSettings.cs
using System;

namespace TailJs.Model;

public record TrackingSettings(
  bool? Promote = null,
  bool? Secondary = null,
  bool? Region = null,
  bool? Clicks = null,
  bool? Impressions = null
)
{
  /// <summary>
  /// Always include in  <see cref="IUserInteractionEvent.Components"/> , also if it is a parent component. By default only the closest component will be included.
  /// 
  /// This does not apply to impression tracking, and has no effect for  <see cref="Content"/>  since these settings automatically promote containing components.
  /// 
  /// Not inherited by child components.
  /// 
  /// HTML attribute: `track-promote`. CSS: `--track-promote: 1/yes/true`.
  /// 
  /// The default value is false.
  /// </summary>
  public bool? Promote { get; set; } = Promote;
  
  /// <summary>
  /// The component will only be tracked with the closest non-secondary component as if the latter had the  <see cref="Promote"/>  flag.
  /// 
  /// This does not apply to impression tracking and has no effect for  <see cref="Content"/> .
  /// 
  /// Not inherited by child components.
  /// 
  /// HTML attribute: `track-secondary`. \ CSS: `--track-secondary: 1/yes/true`.
  /// 
  /// The default value is false.
  /// </summary>
  public bool? Secondary { get; set; } = Secondary;
  
  /// <summary>
  /// Track the visible region occupied by the component or content.
  /// 
  /// Inherited by child components (also if specified on non-component DOM element).
  /// 
  /// HTML attribute: `track-region`. \ CSS: `--track-region: 1/yes/true`.
  /// 
  /// The default value is false.
  /// </summary>
  public bool? Region { get; set; } = Region;
  
  /// <summary>
  /// Track clicks.
  /// 
  /// Inherited by child components (also if specified on non-component DOM element).
  /// 
  /// HTML attribute: `track-clicks`. CSS: `--track-clicks: 1/yes/true`.
  /// 
  /// The default value is true unless in a `&lt;nav&gt;` tag.
  /// </summary>
  public bool? Clicks { get; set; } = Clicks;
  
  /// <summary>
  /// Track impressions, that is, when the component becomes visible in the user&#39;s browser for the first time. This goes well with  <see cref="Region"/> .
  /// 
  /// Not inherited by child components.
  /// 
  /// HTML attribute: `track-impressions`. CSS: `--track-impressions: 1/yes/true`.
  /// 
  /// The default value is false.
  /// </summary>
  public bool? Impressions { get; set; } = Impressions;
}


